Content.Server/Fluids/EntitySystems/PuddleSystem.cs

```diff
@@ -225,9 +228,18 @@ private void OnPuddleSlip(EntityUid uid, PuddleComponent component, ref SlipEven
         _popups.PopupEntity(Loc.GetString("puddle-component-slipped-touch-reaction", ("puddle", uid)),
             args.Slipped, args.Slipped, PopupType.SmallCaution);
 
-        // Take 15% of the puddle solution
-        var splitSol = _solutionContainerSystem.SplitSolution(uid, solution, solution.Volume * 0.15f);
-        _reactive.DoEntityReaction(args.Slipped, splitSol, ReactionMethod.Touch);
+        SpillSolutionOnTarget(solution, args.Slipped, 0.15f);
+        if (_inventory.TryGetContainerSlotEnumerator(args.Slipped, out var containerSlotEnumerator))
```

Comment:
is there a reason this cant be an event that is inventory relayable

Reply:
it can be, but there is no event so why bother

Reply:
yes so make an event :trollface:

Reply:
I can't make this inventory relayed event because clothing is reactive too, meaning it will be raised at least twice
```cs
// TrySplashSpillAt

        var targets = new List<EntityUid>();
        var reactive = new HashSet<Entity<ReactiveComponent>>();
        _lookup.GetEntitiesInRange(coordinates, 1.0f, reactive);

        // Get reactive entities nearby--if there are some, it'll spill a bit on them instead.
        foreach (var ent in reactive)
        {
            // sorry! no overload for returning uid, so .owner must be used
            var owner = ent.Owner;

            if (user != null)
            {
                _adminLogger.Add(LogType.Landed,
                    $"{ToPrettyString(user.Value):user} threw {ToPrettyString(uid):entity} which splashed a solution {SolutionContainerSystem.ToPrettyString(solution):solution} onto {ToPrettyString(owner):target}");
            }

            SpillSolutionOnTarget(solution, owner, _random.NextFloat(0.05f, 0.30f));

            targets.Add(owner);
        }
```

Reply:
You can indicate that you don't want to include entities in containers (which I think includes all bodyparts/clothes) when getting entities in range.

Reply:
fair

Reply:
actually - fake, there is no method to get entities with component AND using flag